### PR TITLE
Allow language parameter in pay button endpoint

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

File: BTCPayServer/Controllers/PublicController.cs [hunk 1]
```diff
@@ -58,7 +59,15 @@ public async Task<IActionResult> PayButtonHandle([FromForm]PayButtonViewModel mo
                 RedirectURL = model.BrowserRedirect,
                 FullNotifications = true
             }, store, HttpContext.Request.GetAbsoluteRoot(), cancellationToken: cancellationToken);
-            return Redirect(invoice.Data.Url);
+            if (string.IsNullOrEmpty(model.Language))
+            {
+                return Redirect(invoice.Data.Url);
+            }
+            var uriBuilder = new UriBuilder(invoice.Data.Url);
+            var paramValues = HttpUtility.ParseQueryString(uriBuilder.Query);
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/PublicController.cs]: don't use this, just add a `string lang = null` parameter in `PayButtonHandle`

File: BTCPayServer/Controllers/PublicController.cs [hunk 2]
```diff
@@ -58,7 +59,15 @@ public async Task<IActionResult> PayButtonHandle([FromForm]PayButtonViewModel mo
...
+            paramValues.Add("lang", model.Language);
+            uriBuilder.Query = paramValues.ToString();
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/PublicController.cs]: I doubt it serialize the dictionarry as you expect.

Comment (NicolasDorier) [reply BTCPayServer/Controllers/PublicController.cs]: it does

**Comment by NicolasDorier** [comment]
What about allowing arbitrary paramters? Instead of `Language` you would have `QueryString`.
And you would just merge this query string to the other one from the invoice?

**Comment by Kukks** [comment]
I can replace Language with Dictionary<string, string> CheckoutAdditionalParameters? and then  you can do <input type='hidden' name='checkoutAdditionalParameters[lang]' value='da-DK'>
this should work as FormData allows these kinds of structures

**Comment by NicolasDorier** [comment]
@Kukks I think `checkoutQueryString` is enough.

**Pull request merged**